### ISSUE TITLE
Path instances with dashes do not work

### ISSUE THREAD

**Issue by bradbishop** [systemd/systemd#5072]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

### systemd version the issue has been seen with

> 5aac0be (1/11 HEAD)

### Used distribution

> Fedora 25

### In case of bug report: Expected behaviour you didn't see

> I'm trying to start a service instance via SYSTEMD_WANTS in a udev rule.  The service is templated on a path.  When the path contains a dash and the template references the unescaped instance, it is not unescaped properly.  

### In case of bug report: Unexpected behaviour you saw
```
$ cat foo@.service
[Service]
ExecStart=/bin/echo %I
$ journalctl -f &
$ systemctl start foo@/bar/baz-foo.service
echo[22021]: /bar/baz/boo
```

### In case of bug report: Steps to reproduce the problem

> systemd-escape -m foo@/bar/baz-foo.service

**Comment by bradbishop** [comment]
https://gerrit.openbmc-project.xyz/#/c/1695/

**Comment by poettering** [comment]
yeah, we currently expect any unit listed in SYSTEMD_WANTS is properly escaped already, and if we notice it isn't we'll do minimal escaping, the same way "systemctl start …" will do it. But you really shouldn't rely on it, as it's not properly reversible, as you just noticed.

A proper fix could be something like adding support for a new SYSTEMD_WANTS_INSTANCE property or so, where you'd specify a template name, and systemd would automatically fill in the device name in escaped form or so.

**Comment by poettering** [comment]
This got merged in the meantime: dcebc9bae4dcc3e844f01558c6127fc0d8745c8e (i.e. v236)

This means when a template name is specified in a udev WANTS property, we'll fill in the properly escaped sysfs path in the instance part of it automatically. I think that's a much nicer approach to the issue at hand. Let's close this hence.
